lib/matplotlib/ticker.py

```diff
@@ -1629,11 +1629,25 @@ def set_params(self, **kwargs):
             str(type(self)))
 
     def __call__(self):
-        """Return the locations of the ticks."""
+        """
+        Return the locations of the ticks.
+
+        Note that the returned ticks depend on the axis assigned to the
+        locator.  If a given locator is used across multiple axises, make sure
+        that
+        """
         # note: some locators return data limits, other return view limits,
         # hence there is no *one* interface to call self.tick_values.
         raise NotImplementedError('Derived must override')
 
+    def _call_with_axis(self, axis):
```

Comment:
This should get a docstring. If I understand it correctly, we always want to call this method instead of `__call__`, which then makes the explicit `self.axis` unused because it is always temporarily overwritten with the passed parameter.

We might choose to make this public as the new API, so people can use it directly (users wanting to evaluate, and locator authors implementing their logic directly here). We could then later deprecate `__call__` and `self.axis`/`set_axis`.

Reply:
Docstring added.  Agreed that this should be the future public API, but I didn't want to commit yet to anything.